Userland/Libraries/LibJS/Bytecode/ASTCodegen.cpp

```diff
@@ -1306,6 +1308,19 @@ static Bytecode::CodeGenerationErrorOr<void> generate_array_binding_pattern_byte
         first = false;
     }
 
+    auto& done_block = generator.make_block();
+    auto& not_done_block = generator.make_block();
+
+    generator.emit<Bytecode::Op::Load>(is_iterator_exhausted_register);
+    generator.emit<Bytecode::Op::JumpConditional>().set_targets(
+        Bytecode::Label { done_block },
+        Bytecode::Label { not_done_block });
+
+    generator.switch_to_basic_block(not_done_block);
+    generator.emit<Bytecode::Op::Load>(iterator_reg);
+    generator.emit<Bytecode::Op::IteratorClose>(Completion::Type::Normal, Optional<Value> {});
+
+    generator.switch_to_basic_block(done_block);
```

Comment:
Now it still seems execution ends[1] when entering the `not_done` block

[1]: By reaching the end of a block without doing any control flow through the terminator

Reply:
Yeah that's clearly wrong. 😅 
And if we fix that, a bunch of tests break... back to the drawing board!